sycl/source/detail/kernel_program_cache.hpp

```diff
@@ -175,6 +232,56 @@ class KernelProgramCache {
   using KernelFastCacheT =
       ::boost::unordered_flat_map<KernelFastCacheKeyT, KernelFastCacheValT>;
 
+  // DS to hold data and functions related to Program cache eviction.
+  struct EvictionListT {
+    // Linked list of cache entries to be evicted in case of cache overflow.
+    std::list<ProgramCacheKeyT> MProgramEvictionList;
+
+    // Mapping between program handle and the iterator to the eviction list.
+    std::unordered_map<ProgramCacheKeyT, std::list<ProgramCacheKeyT>::iterator,
+                       ProgramCacheKeyHash, ProgramCacheKeyEqual>
+        MProgramToEvictionListMap;
```

Comment:
Preferably these members should be private, just to avoid unexpected tampering.

Reply:
Done.